src/snowflake/connector/util_text.py

```diff
@@ -33,6 +43,109 @@ def split_statements(
+    by using special comments. A pure comment line '--SF:<snowflake splitting hints>'
+    All these commands should not be nested
+      (1) '--SF:auto_split off': disable auto split. This should be first cmd in a script file
+      (2) to disable split the sql texts between these two hints
+      '--SF:<<'
+         ...
+      '--SF:>>'
+      (3) to split the blocks before and after this comment '--SF:<<>>'
+    """
+    sqltext = buf.read()
+    sfcmds = []
+    for m in SF_COMMENT_CMD_PATTERN.finditer(sqltext.lower()):
+        sfcmds.append([m.start(), m.end()])
+
+    # stmts to store the split result to return
+    stmts = []
+    is_put_get = []
+    is_auto_split = True
+
+    sqltext_lower = sqltext.lower()
+    for cmd in sfcmds:
+        if is_auto_split and RE_SPLIT_OFF.match(sqltext_lower, cmd[0], cmd[1]):
+            _logger.info("auto_split off")
+            is_auto_split = False
+    starting_pos_of_block = -1  # not in block yet
+    last_pos = 0  # last pos of ending blocks
+
+    # pieces contains tuples [(0|1, str)].
+    # 0 means the string could be split, 1 means it is a whole block
+    pieces = []
+
+    for cmd in sfcmds:
+        if starting_pos_of_block < 0 and RE_BEGIN_BLOCK.match(
+            sqltext_lower, cmd[0], cmd[1]
+        ):
+            starting_pos_of_block = cmd[1]
+            pieces.append([0, sqltext[last_pos : (cmd[0])]])
+        elif starting_pos_of_block > 0 and RE_END_BLOCK.match(
+            sqltext_lower, cmd[0], cmd[1]
+        ):
+            pieces.append([1, sqltext[starting_pos_of_block : (cmd[0])]])
+            last_pos = cmd[1]
+            starting_pos_of_block = -1
+        elif starting_pos_of_block < 0 and RE_SEPARATOR.match(
+            sqltext_lower, cmd[0], cmd[1]
+        ):
+            pieces.append([0, sqltext[last_pos : (cmd[0])]])
+            last_pos = cmd[1]
+
+    if last_pos < len(sqltext):
+        pieces.append([0, sqltext[last_pos:]])
+
+    if not is_auto_split:
+        stmts = [p[1].strip() for p in pieces]
+        is_put_get = [is_put_or_get(stmt) for stmt in stmts]
+    else:  # now apply auto-split to the pieces that were not wrapped by << and >> (piece[0]==1)
+        for piece in pieces:
+            if piece[0] == 0:
+                new_stmts, ret_is_put_get = sep_sqls(piece[1], remove_comments)
+                for idx in range(len(new_stmts)):
+                    stmts.append(new_stmts[idx].strip())
+                    is_put_get.append(ret_is_put_get[idx])
+
+            elif piece[0] == 1:
+                stmt = piece[1].strip()
+                stmts.append(stmt)
+                is_put_get.append(is_put_or_get(stmt))
+
+    for i in range(len(stmts)):
+        yield stmts[i], is_put_get[i]
+
+
+#
+# Below is the old way of splitting statements which allows delimiter
+# and does not support blocks(begin..end) and 'create procedure'
+#
+
+COMMENT_PATTERN_RE = re.compile(r"^\s*\-\-")
+EMPTY_LINE_RE = re.compile(r"^\s*$")
```

Comment:
So these are okay, but only because `re.compile` has caching https://github.com/python/cpython/blob/e2063d6a1ebc3568e90a14ed163fa291b5977ae8/Lib/re.py#L271
I'd not encourage defining constants inside of functions!